### ISSUE TITLE
Graphics: Consider reworking Shader.cs

### ISSUE THREAD

**Issue by bezo97** [NamelessCoding/NamelessFractalRenderer#3]
`Shader.cs` could be reworked. In its current state it really is a wrapper around glProgram, not shaders.. so the naming is confusing.
An observation is that all stages of the pipeline use the same full-screen quad vertex shader, but due to the `Shader` model's implementation, the vertex shaders are copy pasted for each stage even though 1 would be enough.

My suggestion is to rework `Shader.cs` so instead of the super confusing `Shader(vertPath, fragPath)` we would have `VertexShader(path)` and `FragmentShader(path)` and `Program(vertexShader, fragmentShader)` or something similar that makes sense.

This could bring some small performance improvements, and also because the shader-includes won't need to be included in the vertex shaders.

In summary we need to nuke all the `*.vert` files and leave just one, which is read once into a shader and used in all glprograms.
